src/ioctls/kvm_vec.rs

```diff
@@ -0,0 +1,394 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+//
+// Portions Copyright 2017 The Chromium OS Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the THIRD-PARTY file.
+
+use kvm_bindings::__IncompleteArrayField;
+use std::marker::PhantomData;
+use std::mem;
+use std::mem::size_of;
+
+/// Errors associated with the KvmVec struct.
+#[derive(Debug, Clone)]
+pub enum Error {
+    /// The max size has been exceeded
+    SizeLimitExceeded,
+}
+
+// Trait for accessing some properties of certain KVM structures that resemble an array.
+//
+// The kvm API has many structs that resemble the following `T` structure:
+//
+// ```
+// #[repr(C)]
+// struct T {
+//    len: u32,
+//    some_data: u32,
+//    entries: __IncompleteArrayField<F>,
+// }
+// ```
+#[allow(clippy::len_without_is_empty)]
+pub trait KvmArray<F> {
```

Comment:
You can remove this generic type parameter `F` like so:
```
pub trait KvmArray {
    /// Type of kvm struct used as entry
    type Entry;
    
    // the rest is just as before, except methods that were using <F>

    fn entries(&self) -> &__IncompleteArrayField<Self::Entry>;
}
```

While it doesn't make much of difference here, it will help in the definition of `struct KvmVec` (see next comments).